src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,167 @@
+import React from "react";
+import * as RadixDialog from "@radix-ui/react-dialog";
+import styled, { keyframes } from "styled-components";
+import { Button, Icon, Spacer, Title } from "@/components";
+import { EmptyButton } from "../commonElement";
+
+export const Dialog = ({ children, ...props }: RadixDialog.DialogProps) => {
+  return <RadixDialog.Root {...props}>{children}</RadixDialog.Root>;
+};
+
+// Dialog Trigger
+const Trigger = styled(RadixDialog.Trigger)`
+  width: fit-content;
+  background: transparent;
+  border: none;
+  cursor: pointer;
+`;
+
+interface DialogTriggerProps extends RadixDialog.DialogTriggerProps {
+  children: React.ReactNode;
+}
+
+const DialogTrigger = ({ children, ...props }: DialogTriggerProps) => {
+  return (
+    <>
+      <Trigger
+        asChild
+        {...props}
+      >
+        {children}
+      </Trigger>
+    </>
+  );
+};
+
+DialogTrigger.displayName = "DialogTrigger";
+Dialog.Trigger = DialogTrigger;
+
+// Dialog Close
+interface DialogCloseProps extends RadixDialog.DialogCloseProps {
+  label: string;
+}
+
+const DialogClose = ({ label }: DialogCloseProps) => {
```

Comment:
i think we need to pass an `onClick` or a `onSecondaryAction` prop here in case we want the trigger to be external to the Radix Dialog component